more_itertools/more.py

```diff
@@ -2014,7 +2014,7 @@ def difference(iterable, func=sub, *, initial=None):
     if initial is not None:
         first = []
 
-    return chain(first, map(lambda x: func(x[1], x[0]), zip(a, b)))
+    return chain(first, starmap(func, zip(b,a)))
```

Comment:
```suggestion
    return chain(first, starmap(func, zip(b, a)))
```

This will fix the flake8 error.